### ISSUE TITLE
"Checkpoint challenge required..." logger line not working in self.api.send_request()

### ISSUE THREAD

**Issue by leoccyao** [ohld/igbot#1140]
## Please follow the guide below

- Issues submitted without this template format will be **ignored**.
- You will be asked some questions and requested to provide some information, please read them **carefully** and answer completely.
- Put an `x` into all the boxes [ ] relevant to your issue (like so [x]).
- Use the *Preview* tab to see how your issue will actually look like.

---

### Before submitting an issue, make sure you have:
- [x] Updated to the latest version
- [x] Read the [README](https://github.com/instagrambot/instabot/blob/master/README.md) and [docs](https://github.com/instagrambot/instabot/tree/master/docs)
- [x] [Searched](https://github.com/instagrambot/instabot/search?type=Issues) the bugtracker for similar issues including **closed** ones

### Purpose of your issue?
- [x] Bug report (encountered problems/errors)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections requests more details for particular types of issues, you can remove any section (the contents between the triple ---) not applicable to your issue.

---

### For a *bug report*, you **must** include the Python version used, *code* that will reproduce the error, and the *error log/traceback*.

Paste the output of ``python -V`` here:

Python 3.7.4
Running Instabot 0.73.0 in pipenv 2018.11.26

Code:

```python
# Lines 271-277 of api.py (in method send_request_

            elif self.last_json.get(
                "error_type",
                ""
            ) == "checkpoint_challenge_required":
                self.logger.info("Checkpoint challenge required...")
                if ask_for_code is True:
                    solved = self.solve_challenge()

```

Error/Debug Log:
```
Connected to pydev debugger (build 192.6817.19)
2019-11-25 17:38:53,849 - INFO - Instabot Started
Request returns 400 error!
Checkpoint challenge received
0 - Phone
1 - Email
```
---

### Describe your issue

The line "Checkpoint challenge required" was expected to print to console, but did not print. 

This is an issue because: those receiving 400 errors when logging in and therefore "ds_user_id" errors (I received one upon failed login) may not realize that the error is due to a failure to log in.

Can confirm though that checkpoint solving functionality works perfectly (went through it with a successful login!)

Expected output: 
```
Request returns 400 error!
Checkpoint challenge required...
```

**Comment by negswell** [comment]
i think it depends on the logging level that is being set , change logging level to debug in order to get all the logging info .

**Comment by fighterii** [comment]
I'm facing the same issue. As far as I can see, none of the log messages from API.py are logged.

Hmm. logger.warning(and greater: error/critical) are logged (but not in the specified Logger Format with debugLevel) to stdout, but no logger.info messages. Very Strange behavior.

@negswell as you can see in @leoccyao code example the line is `logger.info`, so it has nothing to do with log level.

**Comment by fighterii** [comment]
@ohld :
Something is inconsistent in the code:
- In the constructor of API you create self.logger for API (api.py line 63-88). There the levels are set and file/streamhandler is added.
- This logger instance is than passed to bot (bot.py line 267)
- When the login is called (api.py set_user in line 185) the api.logger is overwritten (api.py line 96).
=> Now we have two different loggers (one in bot.logger with 2 handlers and levels; one in api.logger with 0 handlers)

You think it is really necessary to have the logger name including the user_id/username? What about one Logger "[instabot]"?

**Comment by fighterii** [comment]
Logging is fixed with PR #1205 so we could close this issue @ohld
